### ISSUE TITLE
private_dns_cname_record TTL is set to be minimum 1 when it is possible to set TTL to 0 on the API

### ISSUE THREAD

**Issue by maonat** [hashicorp/terraform-provider-azurerm#23917]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.3

### AzureRM Provider Version

3.80.0

### Affected Resource(s)/Data Source(s)

azurerm_private_dns_cname_record

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_private_dns_zone" "example" {
  name                = "mydomain.com"
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_private_dns_cname_record" "example" {
  name                = "test"
  zone_name           = azurerm_private_dns_zone.example.name
  resource_group_name = azurerm_resource_group.example.name
  ttl                 = 0
  record              = "contoso.com"
}
```

### Debug Output/Panic Output

```shell
│ Error: expected ttl to be in the range (1 - 2147483647), got 0
│ 
│   with azurerm_private_dns_cname_record.example,
│   on main.tf line 43, in resource "azurerm_private_dns_cname_record" "example":
│   43:   ttl                 = 0
│
```

### Expected Behaviour

TTL set to 0 for the record test.mydomain.com

### Actual Behaviour

Error on apply

**Comment by maonat** [comment]
Below the line with the issue:
https://github.com/hashicorp/terraform-provider-azurerm/blob/ba3eb6e3eec17560bf6985478b018eec2863315f/internal/services/privatedns/private_dns_cname_record_resource.go#L78C56-L78C56

**Comment by maonat** [comment]
Piece of code to fix from 1 to 0:
https://github.com/hashicorp/terraform-provider-azurerm/blob/ba3eb6e3eec17560bf6985478b018eec2863315f/internal/services/privatedns/private_dns_cname_record_resource.go#L75-L79

**Comment by stephybun** [comment]
Closed by #23918
